packages/compartment-mapper/src/link.js

```diff
@@ -175,6 +262,41 @@ export const mapParsers = (
   );
 };
 
+/**
+ * @param {Record<string, Language>} languageForExtension
+ * @param {Record<string, string>} languageForModuleSpecifier - In a rare case, the type of a module
+ * is implied by package.json and should not be inferred from its extension.
+ * @param {Record<string, ParserImplementation>} parserForLanguage
+ * @param {SyncModuleTransforms} moduleTransforms
+ * @returns {ParseFn}
+ */
+export const mapParsersSync = (
```

Comment:
should probably just change mapParsers to handle both `SyncModuleTransforms` and `ModuleTransforms`...?